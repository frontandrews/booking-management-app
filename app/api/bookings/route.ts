import { NextResponse, NextRequest } from 'next/server';
import axios from 'axios';
import { API_URL } from '@/constants';
import { isValid, parseISO } from 'date-fns';
import { handleAxiosError } from '@/utils';

export async function POST(request: NextRequest) {
  try {
    const { name, propertyId, startDate, endDate, pricePerDay } =
      await request.json();

    if (!name || !propertyId || !startDate || !endDate || !pricePerDay) {
      return NextResponse.json(
        {
          message:
            'name, propertyId, startDate, endDate, and pricePerDay are required',
        },
        { status: 400 },
      );
    }

    if (!isValid(parseISO(startDate)) || !isValid(parseISO(endDate))) {
      return NextResponse.json(
        { message: 'Invalid date format' },
        { status: 400 },
      );
    }

    const response = await axios.post(`${API_URL}/bookings`, {
      name,
      propertyId,
      startDate,
      endDate,
      pricePerDay,
    });
    return NextResponse.json(response.data, { status: 201 });
  } catch (error: any) {
    return handleAxiosError(error);
  }
}

export async function GET() {
  try {
    const response = await axios.get(`${API_URL}/bookings`);
    return NextResponse.json(response.data, { status: 200 });
  } catch (error: any) {
    return handleAxiosError(error);
  }
}
