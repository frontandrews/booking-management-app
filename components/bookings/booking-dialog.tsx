import { useState, ReactNode } from 'react';
import { Button } from '@/components/ui/button';
import {
  Dialog,
  DialogContent,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
} from '@/components/ui/dialog';
import BookingForm from './booking-form';
import { Plus } from 'lucide-react';

interface BookingDialogProps {
  bookingId?: number;
  buttonClassName?: string;
  children?: ReactNode;
}

export function BookingDialog({
  bookingId,
  buttonClassName,
  children,
}: BookingDialogProps) {
  const [isOpen, setIsOpen] = useState(false);
  const handleClose = () => setIsOpen(false);

  return (
    <Dialog open={isOpen} onOpenChange={setIsOpen}>
      <DialogTrigger asChild>
        {children ? (
          <div className={buttonClassName} onClick={() => setIsOpen(true)}>
            {children}
          </div>
        ) : (
          <Button
            variant="default"
            size="sm"
            className={`ml-2 ${buttonClassName}`}
            onClick={() => setIsOpen(true)}
          >
            <Plus className="w-4 h-4 mr-2" />
            Reservation
          </Button>
        )}
      </DialogTrigger>
      <DialogContent className="sm:max-w-[425px] pb-6">
        <DialogHeader className="px-6 pt-4">
          <DialogTitle>
            {bookingId ? 'Edit Reservation' : 'New Reservation'}
          </DialogTitle>
        </DialogHeader>
        <BookingForm bookingId={bookingId} onClose={handleClose} />
      </DialogContent>
    </Dialog>
  );
}
