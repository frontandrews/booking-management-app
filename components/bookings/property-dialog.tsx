import { useState } from 'react';
import { Button } from '@/components/ui/button';
import {
  Dialog,
  DialogContent,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
} from '@/components/ui/dialog';
import PropertyForm from './property-form';
import { EllipsisVertical, Plus } from 'lucide-react';

interface PropertyDialogProps {
  propertyId?: number;
}

export function PropertyDialog({ propertyId }: PropertyDialogProps) {
  const [isOpen, setIsOpen] = useState(false);
  const handleClose = () => setIsOpen(false);

  return (
    <Dialog open={isOpen} onOpenChange={setIsOpen}>
      <DialogTrigger asChild>
        {propertyId ? (
          <button className="p-2 border rounded hover:bg-gray-100">
            <EllipsisVertical className="w-4 h-4" />
          </button>
        ) : (
          <Button variant="default" size="sm" className="text-center">
            <Plus className="w-4 h-4" />
          </Button>
        )}
      </DialogTrigger>
      <DialogContent className="sm:max-w-[425px] pb-6">
        <DialogHeader className="px-6 pt-4">
          <DialogTitle>
            {propertyId ? 'Edit Property' : 'Add Property'}
          </DialogTitle>
        </DialogHeader>
        <PropertyForm propertyId={propertyId} onClose={handleClose} />
      </DialogContent>
    </Dialog>
  );
}
