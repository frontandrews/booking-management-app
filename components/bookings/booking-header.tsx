import React from 'react';
import { ChevronLeftIcon, ChevronRightIcon } from '@heroicons/react/20/solid';
import { format } from 'date-fns';
import { BookingDialog } from './booking-dialog';

interface BookingHeaderProps {
  currentMonth: Date;
  onPreviousMonth: () => void;
  onNextMonth: () => void;
  onToday: () => void;
}

export const BookingHeader = ({
  currentMonth,
  onPreviousMonth,
  onNextMonth,
  onToday,
}: BookingHeaderProps) => {
  return (
    <header className="flex flex-col items-center justify-between border-b border-gray-200 py-4 lg:flex-row lg:flex-none">
      <h1 className="text-2xl font-semibold leading-6 text-gray-900 mb-4 lg:mb-0">
        <time dateTime={format(currentMonth, 'yyyy-MM')}>
          {format(currentMonth, 'MMMM yyyy')}
        </time>
      </h1>
      <div className="flex flex-col items-center lg:flex-row lg:space-x-4">
        <div className="relative flex items-center rounded-md bg-white shadow-sm mb-4 lg:mb-0 md:items-stretch">
          <button
            type="button"
            className="flex h-9 w-12 items-center justify-center rounded-l-md border-y border-l border-gray-300 pr-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pr-0 md:hover:bg-gray-50"
            onClick={onPreviousMonth}
          >
            <span className="sr-only">Previous month</span>
            <ChevronLeftIcon className="h-5 w-5" aria-hidden="true" />
          </button>
          <button
            type="button"
            className="border-x flex h-9 items-center justify-center border-y border-gray-300 px-3.5 text-sm font-semibold text-gray-900 hover:bg-gray-50 focus:relative md:block"
            onClick={onToday}
          >
            Today
          </button>
          <span className="relative -mx-px h-9 w-px bg-gray-300 md:hidden" />
          <button
            type="button"
            className="flex h-9 w-12 items-center justify-center rounded-r-md border-y border-r border-gray-300 pl-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pl-0 md:hover:bg-gray-50"
            onClick={onNextMonth}
          >
            <span className="sr-only">Next month</span>
            <ChevronRightIcon className="h-5 w-5" aria-hidden="true" />
          </button>
        </div>
        <BookingDialog />
      </div>
    </header>
  );
};
