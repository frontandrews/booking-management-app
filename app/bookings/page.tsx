'use client';

import { useEffect, useRef, useState } from 'react';
import { PropertyDialog } from '@/components/bookings/property-dialog';
import PropertyList from '@/components/bookings/property-list';
import SectionHeading from '@/components/ui/section-heading';
import BookingView from '@/components/bookings/bookings-view';
import {
  fetchProperties,
  setSelectedPropertyId,
} from '@/redux/features/property/slice';
import EmptyState from '@/components/ui/empty-state';
import { ThreeDots } from 'react-loader-spinner';
import { fetchBookings } from '@/redux/features/booking/slice';
import { PropertyStats } from '@/components/bookings/property-stats';
import { useAppDispatch, useAppSelector } from '@/redux/hooks';

export default function BookingsPage() {
  const dispatch = useAppDispatch();
  const selectedPropertyId = useAppSelector(
    (state) => state.property.selectedPropertyId,
  );
  const bookings = useAppSelector((state) => state.booking.bookings);
  const properties = useAppSelector((state) => state.property.properties);
  const loading = useAppSelector((state) => state.property.loading);

  const [initialLoading, setInitialLoading] = useState(true);
  const hasAutoSelected = useRef(false);

  useEffect(() => {
    if (properties.length === 0) {
      dispatch(fetchProperties()).then(() => setInitialLoading(false));
    } else {
      setInitialLoading(false);
    }
  }, [properties.length, dispatch]);

  // Auto select first property
  useEffect(() => {
    if (!hasAutoSelected.current && properties.length > 0) {
      const id = properties[0].id;
      if (id) {
        if (id !== selectedPropertyId) {
          dispatch(setSelectedPropertyId(id));
        }
        dispatch(fetchBookings(id));
      }
      hasAutoSelected.current = true;
    }
  }, [properties, selectedPropertyId, dispatch]);

  return (
    <div className="container mx-auto">
      <SectionHeading title="Dashboard" />

      {initialLoading || loading ? (
        <div className="flex justify-center py-10">
          <ThreeDots
            visible={true}
            height="80"
            width="80"
            color="black"
            radius="9"
            ariaLabel="three-dots-loading"
          />
        </div>
      ) : properties.length === 0 ? (
        <EmptyState
          title="No properties found"
          description="Create a new property to get started."
        >
          <PropertyDialog />
        </EmptyState>
      ) : (
        <>
          <div className="grid grid-cols-12 gap-4">
            <div className="col-span-12 lg:col-span-4">
              <PropertyList />
            </div>
            <div className="col-span-12 lg:col-span-8">
              <BookingView bookings={bookings} />
            </div>
          </div>
          <PropertyStats />
        </>
      )}
    </div>
  );
}
