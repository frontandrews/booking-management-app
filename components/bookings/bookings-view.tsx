import React, { useState, useEffect } from 'react';
import {
  startOfMonth,
  endOfMonth,
  startOfWeek,
  endOfWeek,
  addDays,
  addMonths,
  subMonths,
} from 'date-fns';
import { BookingHeader } from './booking-header';
import { Calendar } from './calendar';
import { Day, BookingViewProps } from './types';

const BookingView = ({ bookings = [] }: BookingViewProps) => {
  const [currentMonth, setCurrentMonth] = useState<Date>(new Date());
  const [days, setDays] = useState<Day[]>([]);

  useEffect(() => {
    const generateDays = (date: Date): Day[] => {
      const startDate = startOfWeek(startOfMonth(date));
      const endDate = endOfWeek(endOfMonth(date));
      const days: Day[] = [];
      let day = startDate;

      while (day <= endDate) {
        days.push(day);
        day = addDays(day, 1);
      }
      return days;
    };
    setDays(generateDays(currentMonth));
  }, [currentMonth]);

  const handlePreviousMonth = () => {
    setCurrentMonth(subMonths(currentMonth, 1));
  };

  const handleNextMonth = () => {
    setCurrentMonth(addMonths(currentMonth, 1));
  };

  const handleToday = () => {
    setCurrentMonth(new Date());
  };

  return (
    <div className="lg:flex lg:h-full lg:flex-col px-4 pb-4 border">
      <BookingHeader
        currentMonth={currentMonth}
        onPreviousMonth={handlePreviousMonth}
        onNextMonth={handleNextMonth}
        onToday={handleToday}
      />
      <Calendar days={days} currentMonth={currentMonth} bookings={bookings} />
    </div>
  );
};

export default BookingView;
