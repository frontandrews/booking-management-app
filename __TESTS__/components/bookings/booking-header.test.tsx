import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import { BookingHeader } from '@/components/bookings/booking-header';
import { format } from 'date-fns';

describe('BookingHeader Component', () => {
  const mockOnPreviousMonth = jest.fn();
  const mockOnNextMonth = jest.fn();
  const mockOnToday = jest.fn();
  const currentMonth = new Date(2024, 6, 15); // July 2024

  beforeEach(() => {
    render(
      <BookingHeader
        currentMonth={currentMonth}
        onPreviousMonth={mockOnPreviousMonth}
        onNextMonth={mockOnNextMonth}
        onToday={mockOnToday}
      />,
    );
  });

  it('renders the current month', () => {
    const formattedMonth = format(currentMonth, 'MMMM yyyy');
    expect(screen.getByText(formattedMonth)).toBeInTheDocument();

    const { container } = render(
      <BookingHeader
        currentMonth={currentMonth}
        onPreviousMonth={mockOnPreviousMonth}
        onNextMonth={mockOnNextMonth}
        onToday={mockOnToday}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('calls onPreviousMonth when the Previous month button is clicked', () => {
    const previousButton = screen.getByRole('button', {
      name: /Previous month/i,
    });
    fireEvent.click(previousButton);
    expect(mockOnPreviousMonth).toHaveBeenCalledTimes(1);

    const { container } = render(
      <BookingHeader
        currentMonth={currentMonth}
        onPreviousMonth={mockOnPreviousMonth}
        onNextMonth={mockOnNextMonth}
        onToday={mockOnToday}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('calls onNextMonth when the Next month button is clicked', () => {
    const nextButton = screen.getByRole('button', { name: /Next month/i });
    fireEvent.click(nextButton);
    expect(mockOnNextMonth).toHaveBeenCalledTimes(1);

    const { container } = render(
      <BookingHeader
        currentMonth={currentMonth}
        onPreviousMonth={mockOnPreviousMonth}
        onNextMonth={mockOnNextMonth}
        onToday={mockOnToday}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('calls onToday when the Today button is clicked', () => {
    const todayButton = screen.getByRole('button', { name: /Today/i });
    fireEvent.click(todayButton);
    expect(mockOnToday).toHaveBeenCalledTimes(1);

    const { container } = render(
      <BookingHeader
        currentMonth={currentMonth}
        onPreviousMonth={mockOnPreviousMonth}
        onNextMonth={mockOnNextMonth}
        onToday={mockOnToday}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the BookingDialog component', () => {
    expect(screen.getByTestId('booking-dialog')).toBeInTheDocument();

    const { container } = render(
      <BookingHeader
        currentMonth={currentMonth}
        onPreviousMonth={mockOnPreviousMonth}
        onNextMonth={mockOnNextMonth}
        onToday={mockOnToday}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
