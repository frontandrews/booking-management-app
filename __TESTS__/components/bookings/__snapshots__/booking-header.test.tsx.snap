// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingHeader Component calls onNextMonth when the Next month button is clicked 1`] = `
<div>
  <header
    class="flex flex-col items-center justify-between border-b border-gray-200 py-4 lg:flex-row lg:flex-none"
    data-testid="booking-dialog"
  >
    <h1
      class="text-2xl font-semibold leading-6 text-gray-900 mb-4 lg:mb-0"
    >
      <time
        datetime="2024-07"
      >
        July 2024
      </time>
    </h1>
    <div
      class="flex flex-col items-center lg:flex-row lg:space-x-4"
    >
      <div
        class="relative flex items-center rounded-md bg-white shadow-sm mb-4 lg:mb-0 md:items-stretch"
      >
        <button
          class="flex h-9 w-12 items-center justify-center rounded-l-md border-y border-l border-gray-300 pr-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pr-0 md:hover:bg-gray-50"
          type="button"
        >
          <span
            class="sr-only"
          >
            Previous month
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.78 5.22a.75.75 0 0 1 0 1.06L8.06 10l3.72 3.72a.75.75 0 1 1-1.06 1.06l-4.25-4.25a.75.75 0 0 1 0-1.06l4.25-4.25a.75.75 0 0 1 1.06 0Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <button
          class="border-x flex h-9 items-center justify-center border-y border-gray-300 px-3.5 text-sm font-semibold text-gray-900 hover:bg-gray-50 focus:relative md:block"
          type="button"
        >
          Today
        </button>
        <span
          class="relative -mx-px h-9 w-px bg-gray-300 md:hidden"
        />
        <button
          class="flex h-9 w-12 items-center justify-center rounded-r-md border-y border-r border-gray-300 pl-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pl-0 md:hover:bg-gray-50"
          type="button"
        >
          <span
            class="sr-only"
          >
            Next month
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8.22 5.22a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.75.75 0 0 1-1.06-1.06L11.94 10 8.22 6.28a.75.75 0 0 1 0-1.06Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <button
        aria-controls="radix-:rf:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3 ml-2 undefined"
        data-state="closed"
        type="button"
      >
        <svg
          class="lucide lucide-plus w-4 h-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
        Reservation
      </button>
    </div>
  </header>
</div>
`;

exports[`BookingHeader Component calls onPreviousMonth when the Previous month button is clicked 1`] = `
<div>
  <header
    class="flex flex-col items-center justify-between border-b border-gray-200 py-4 lg:flex-row lg:flex-none"
    data-testid="booking-dialog"
  >
    <h1
      class="text-2xl font-semibold leading-6 text-gray-900 mb-4 lg:mb-0"
    >
      <time
        datetime="2024-07"
      >
        July 2024
      </time>
    </h1>
    <div
      class="flex flex-col items-center lg:flex-row lg:space-x-4"
    >
      <div
        class="relative flex items-center rounded-md bg-white shadow-sm mb-4 lg:mb-0 md:items-stretch"
      >
        <button
          class="flex h-9 w-12 items-center justify-center rounded-l-md border-y border-l border-gray-300 pr-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pr-0 md:hover:bg-gray-50"
          type="button"
        >
          <span
            class="sr-only"
          >
            Previous month
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.78 5.22a.75.75 0 0 1 0 1.06L8.06 10l3.72 3.72a.75.75 0 1 1-1.06 1.06l-4.25-4.25a.75.75 0 0 1 0-1.06l4.25-4.25a.75.75 0 0 1 1.06 0Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <button
          class="border-x flex h-9 items-center justify-center border-y border-gray-300 px-3.5 text-sm font-semibold text-gray-900 hover:bg-gray-50 focus:relative md:block"
          type="button"
        >
          Today
        </button>
        <span
          class="relative -mx-px h-9 w-px bg-gray-300 md:hidden"
        />
        <button
          class="flex h-9 w-12 items-center justify-center rounded-r-md border-y border-r border-gray-300 pl-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pl-0 md:hover:bg-gray-50"
          type="button"
        >
          <span
            class="sr-only"
          >
            Next month
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8.22 5.22a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.75.75 0 0 1-1.06-1.06L11.94 10 8.22 6.28a.75.75 0 0 1 0-1.06Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <button
        aria-controls="radix-:r9:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3 ml-2 undefined"
        data-state="closed"
        type="button"
      >
        <svg
          class="lucide lucide-plus w-4 h-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
        Reservation
      </button>
    </div>
  </header>
</div>
`;

exports[`BookingHeader Component calls onToday when the Today button is clicked 1`] = `
<div>
  <header
    class="flex flex-col items-center justify-between border-b border-gray-200 py-4 lg:flex-row lg:flex-none"
    data-testid="booking-dialog"
  >
    <h1
      class="text-2xl font-semibold leading-6 text-gray-900 mb-4 lg:mb-0"
    >
      <time
        datetime="2024-07"
      >
        July 2024
      </time>
    </h1>
    <div
      class="flex flex-col items-center lg:flex-row lg:space-x-4"
    >
      <div
        class="relative flex items-center rounded-md bg-white shadow-sm mb-4 lg:mb-0 md:items-stretch"
      >
        <button
          class="flex h-9 w-12 items-center justify-center rounded-l-md border-y border-l border-gray-300 pr-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pr-0 md:hover:bg-gray-50"
          type="button"
        >
          <span
            class="sr-only"
          >
            Previous month
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.78 5.22a.75.75 0 0 1 0 1.06L8.06 10l3.72 3.72a.75.75 0 1 1-1.06 1.06l-4.25-4.25a.75.75 0 0 1 0-1.06l4.25-4.25a.75.75 0 0 1 1.06 0Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <button
          class="border-x flex h-9 items-center justify-center border-y border-gray-300 px-3.5 text-sm font-semibold text-gray-900 hover:bg-gray-50 focus:relative md:block"
          type="button"
        >
          Today
        </button>
        <span
          class="relative -mx-px h-9 w-px bg-gray-300 md:hidden"
        />
        <button
          class="flex h-9 w-12 items-center justify-center rounded-r-md border-y border-r border-gray-300 pl-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pl-0 md:hover:bg-gray-50"
          type="button"
        >
          <span
            class="sr-only"
          >
            Next month
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8.22 5.22a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.75.75 0 0 1-1.06-1.06L11.94 10 8.22 6.28a.75.75 0 0 1 0-1.06Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <button
        aria-controls="radix-:rl:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3 ml-2 undefined"
        data-state="closed"
        type="button"
      >
        <svg
          class="lucide lucide-plus w-4 h-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
        Reservation
      </button>
    </div>
  </header>
</div>
`;

exports[`BookingHeader Component renders the BookingDialog component 1`] = `
<div>
  <header
    class="flex flex-col items-center justify-between border-b border-gray-200 py-4 lg:flex-row lg:flex-none"
    data-testid="booking-dialog"
  >
    <h1
      class="text-2xl font-semibold leading-6 text-gray-900 mb-4 lg:mb-0"
    >
      <time
        datetime="2024-07"
      >
        July 2024
      </time>
    </h1>
    <div
      class="flex flex-col items-center lg:flex-row lg:space-x-4"
    >
      <div
        class="relative flex items-center rounded-md bg-white shadow-sm mb-4 lg:mb-0 md:items-stretch"
      >
        <button
          class="flex h-9 w-12 items-center justify-center rounded-l-md border-y border-l border-gray-300 pr-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pr-0 md:hover:bg-gray-50"
          type="button"
        >
          <span
            class="sr-only"
          >
            Previous month
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.78 5.22a.75.75 0 0 1 0 1.06L8.06 10l3.72 3.72a.75.75 0 1 1-1.06 1.06l-4.25-4.25a.75.75 0 0 1 0-1.06l4.25-4.25a.75.75 0 0 1 1.06 0Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <button
          class="border-x flex h-9 items-center justify-center border-y border-gray-300 px-3.5 text-sm font-semibold text-gray-900 hover:bg-gray-50 focus:relative md:block"
          type="button"
        >
          Today
        </button>
        <span
          class="relative -mx-px h-9 w-px bg-gray-300 md:hidden"
        />
        <button
          class="flex h-9 w-12 items-center justify-center rounded-r-md border-y border-r border-gray-300 pl-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pl-0 md:hover:bg-gray-50"
          type="button"
        >
          <span
            class="sr-only"
          >
            Next month
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8.22 5.22a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.75.75 0 0 1-1.06-1.06L11.94 10 8.22 6.28a.75.75 0 0 1 0-1.06Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <button
        aria-controls="radix-:rr:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3 ml-2 undefined"
        data-state="closed"
        type="button"
      >
        <svg
          class="lucide lucide-plus w-4 h-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
        Reservation
      </button>
    </div>
  </header>
</div>
`;

exports[`BookingHeader Component renders the current month 1`] = `
<div>
  <header
    class="flex flex-col items-center justify-between border-b border-gray-200 py-4 lg:flex-row lg:flex-none"
    data-testid="booking-dialog"
  >
    <h1
      class="text-2xl font-semibold leading-6 text-gray-900 mb-4 lg:mb-0"
    >
      <time
        datetime="2024-07"
      >
        July 2024
      </time>
    </h1>
    <div
      class="flex flex-col items-center lg:flex-row lg:space-x-4"
    >
      <div
        class="relative flex items-center rounded-md bg-white shadow-sm mb-4 lg:mb-0 md:items-stretch"
      >
        <button
          class="flex h-9 w-12 items-center justify-center rounded-l-md border-y border-l border-gray-300 pr-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pr-0 md:hover:bg-gray-50"
          type="button"
        >
          <span
            class="sr-only"
          >
            Previous month
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.78 5.22a.75.75 0 0 1 0 1.06L8.06 10l3.72 3.72a.75.75 0 1 1-1.06 1.06l-4.25-4.25a.75.75 0 0 1 0-1.06l4.25-4.25a.75.75 0 0 1 1.06 0Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <button
          class="border-x flex h-9 items-center justify-center border-y border-gray-300 px-3.5 text-sm font-semibold text-gray-900 hover:bg-gray-50 focus:relative md:block"
          type="button"
        >
          Today
        </button>
        <span
          class="relative -mx-px h-9 w-px bg-gray-300 md:hidden"
        />
        <button
          class="flex h-9 w-12 items-center justify-center rounded-r-md border-y border-r border-gray-300 pl-1 text-gray-400 hover:text-gray-500 focus:relative md:w-9 md:pl-0 md:hover:bg-gray-50"
          type="button"
        >
          <span
            class="sr-only"
          >
            Next month
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8.22 5.22a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.75.75 0 0 1-1.06-1.06L11.94 10 8.22 6.28a.75.75 0 0 1 0-1.06Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <button
        aria-controls="radix-:r3:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3 ml-2 undefined"
        data-state="closed"
        type="button"
      >
        <svg
          class="lucide lucide-plus w-4 h-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
        Reservation
      </button>
    </div>
  </header>
</div>
`;
